src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/json/JsonConvert.java

```diff
@@ -71,26 +77,49 @@ private static Path startApartments() {
    */
   @SuppressWarnings("serial")
   public static List<Apartment> jsonToApartments(Path jsonPath) throws IOException {
+    JsonbConfig config = new JsonbConfig().withAdapters(JsonConvert.getAdapter());
     String jsonString = Files.readString(jsonPath);
-    List<Apartment.Builder> apartmentsBuild;
     List<Apartment> apartments = new ArrayList<>();
     LOGGER.info("Create ArrayList of Apartment");
-
-    try (Jsonb jsonb = JsonbBuilder.create()) {
+    try (Jsonb jsonb = JsonbBuilder.create(config)) {
       LOGGER.info("Create Json builder");
-      apartmentsBuild =
+      apartments =
           jsonb.fromJson(
-              jsonString, new ArrayList<Apartment.Builder>() {}.getClass().getGenericSuperclass());
+              jsonString,
+              new ArrayList<Apartment>() {
+                /**/
+              }.getClass().getGenericSuperclass());
     } catch (Exception e) {
       throw new IllegalArgumentException(e);
     }
-
-    for (Builder apartmentToBuild : apartmentsBuild) {
-      apartments.add(apartmentToBuild.build());
-    }
     return apartments;
   }
 
+  /** A method that implement a {@link JsonbAdapter} to avoid the use of ApartmentBuilder Code */
+  public static JsonbAdapter<Apartment, JsonObject> getAdapter() {
+    return new JsonbAdapter<>() {
+      @Override
+      public JsonObject adaptToJson(Apartment obj) {
+        throw new UnsupportedOperationException("This function should not be called");
+      }
+
+      /**
+       * Convert an Apartment json into an Apartment object
+       *
+       * @param obj Apartment json content
+       * @return the Apartment built using <code>obj</code>
+       * @throws Exception raises an Exception thrown by JsonBuilder
+       */
+      @Override
+      public Apartment adaptFromJson(JsonObject obj) throws Exception {
+        checkNotNull(obj, "The JsonObject 'obj' can't be null");
+        try (Jsonb jsonb = JsonbBuilder.create()) {
+          return jsonb.fromJson(obj.toString(), Apartment.Builder.class).build();
+        }
```

Comment:
Il faut éviter de faire appel à Jsonb alors qu’on est dans une méthode censée aider Jsonb à traiter le cas. (Voir remarque ci-dessus.)

Reply:
Fix in 2fd4cc3729016854db9dc872f869b9b221bd4ac2